src/edge.js

```diff
@@ -320,7 +324,32 @@ export class DocRoom {
     // eslint-disable-next-line no-console
     console.log(`Setting up WSConnection for ${docName} with auth(${webSocket.auth
       ? webSocket.auth.substring(0, webSocket.auth.indexOf(' ')) : 'none'})`);
-    const timingData = await setupWSConnection(webSocket, docName, this.env, this.storage);
+
+    const timingData = new Map();
+
+    let ydoc = this.docs.get(docName);
+    if (!ydoc) {
+      // get doc, initialize if it does not exist yet
+      ydoc = await createYDoc(
+        docName,
+        webSocket,
+        this.env,
+        this.storage,
+        timingData,
+        this.docs,
+        true,
+      );
+      this.docs.set(docName, ydoc);
+    }
+
+    webSocket.addEventListener('close', () => {
```

Comment:
I'm not sure this is correct. Wouldn't this basically remove the doc if a connection goes away?

Reply:
On connection close, the doc is removed from the map ("local cache"). This is the same logic as before, just moved the code here to have access to the cache.